src/highlight/tests.rs

```diff
@@ -866,10 +866,13 @@ mod file_tester_tests {
         assert!(!result);
 
         // Can't redirect from an unreadable file.
-        fs::set_permissions(&file_path, Permissions::from_mode(0o200)).unwrap();
-        let result = tester.test_redirection_target(L!("file.txt"), RedirectionMode::input);
-        assert!(!result);
-        fs::set_permissions(&file_path, Permissions::from_mode(0o600)).unwrap();
+        #[cfg(not(cygwin))] // Can't mark a file write-only on MSYS, this may work on true Cygwin
```

Comment:
> Note: Cygwin by default does use "acl", which may allow the check to pass (unverified).

`build.rs` currently defines `cfg(cygwin)` as

	env_var("CARGO_CFG_TARGET_OS").unwrap() == "cygwin"

Perhaps `build.rs` can detect the acl/noacl state of the current file system,
and base our check on that?
I.e. the equivalent of

	set mountpoint $(df . --output=source | tail -1)
	mount | string match -r ^$mountpoint.\* | grep noacl

Reply:
> Are tests expected to be run only in the build environment?

Not sure right now, but this might already be assumed elsewhere,
like in our workspace-level `build.rs`.
But yeah, querying in build.rs doesn't seem right,
but we could do it at test run time instead.

> would cause different people potentially seeing different test results

yeah but that's okay - people who enable ACLs *should* see a failure,
so we can actually address it

If we could detect when we're building for msys2 (as opposed to cygwin),
we could skip it only for that case. That would be better,
though not as good as probing for ACL support.

It's fine to leave it as you wrote for now,
just saying how we can improve this.